packages/jest-cli/src/prepareStackTrace.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+'use strict';
+
+// filter for noisy stack trace lines
+const FILTER_NOISE =
+  /(\/|\\)jest(-.*?)?(\/|\\)(vendor|build|node_modules|packages)(\/|\\)/;
+
+module.exports = (context: {Error: Class<Error>}) => {
+  const Error = context.Error;
+  Error.prepareStackTrace = (error: Error, stacktrace: CallSite[]): mixed => {
+    const filteredStacktrace = stacktrace.filter(
+      (callsite: CallSite, index: number) => {
+        return index === 0 || !FILTER_NOISE.test(callsite.getFileName() || '');
+      }
+    ).map(callsite => {
+      const what = callsite.getFunctionName() || '<anonymous>';
+      const filename = callsite.getFileName() || '';
+      const line = callsite.getLineNumber() || '0';
+      const column = callsite.getColumnNumber() || '0';
+
+      return `at ${what} (${filename}:${line}:${column})`;
+    });
+    
+    return `${error.name}: ${error.message}\n${filteredStacktrace.join('\n')}`;
```

Comment:
Nice use of prepareStackTrace. I don't quite see why we are doing all this work only to turn it back into a string. I was hoping we could change "failureMessages" into a structure like this:

```
type FailureMessage = {
  message: string,
  stack: Array<StackItem>,
};

type FailureMessages = Array<FailureMessage>;

type Stack = {|
  … // All the useful stack data
|};
```

and then have the rest of Jest operate on these data structures rather than doing more string stack parsing everywhere.